pkg/daemon/daemon.go

```diff
@@ -573,6 +574,26 @@ func (dn *Daemon) RunOnceFrom(onceFrom string, skipReboot bool) error {
 	return errors.New("unsupported onceFrom type provided")
 }
 
+// synthesizeCurrentState creates a machineconfig object which describes
+// what we see on the node.
+func (dn *Daemon) synthesizeCurrentState() (*mcfgv1.MachineConfig, error) {
+	// Start with an empty config, then add our *booted* osImageURL to
+	// it, reflecting the current machine state.
+	config := canonicalizeEmptyMC(nil)
+	config.Spec.OSImageURL = dn.bootedOSImageURL
+	// Also inject the current fips state, which was handled before we run.
+	content, err := ioutil.ReadFile(fipsFile)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading FIPS file at %s: %w", fipsFile, err)
```

Comment:
`checkFIPS` has
```
		if os.IsNotExist(err) {
			// we just exit cleanly if we're not even on linux
			glog.Infof("no %s on this system, skipping FIPS check", fipsFile)
			return nil
		}
```
I think it would make more sense if we moved the FIPS code to a getFIPS() function so we handle cases like that consistently. And then maybe instead of my PR just have
```
if desired.Spec.FIPS != getFIPS() {
  return fmt.Errorf(...)
}
```

Reply:
Agree a factored out `getFIPS()` function would make sense, though I don't understand how we could reach that particular bit since the MCD only runs on Linux.  (maybe somehow the unit tests reach it when run from MacOS e.g.?)